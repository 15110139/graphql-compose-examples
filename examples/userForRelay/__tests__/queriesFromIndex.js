import { graphql } from 'graphql';
import meta from '../index';

function findQueryByTitle(str) {
  const queryConfig = meta.queries.find(o => o.title === str);
  if (queryConfig && queryConfig.query) {
    return queryConfig.query;
  }
  return 'query not found';
}

describe('userForRelay > queries', () => {
  const alwaysSameResultTitles = [
    'Relay node',
    'Relay Connection',
  ];
  alwaysSameResultTitles.forEach((title) => {
    it(title, async () => {
      const result = await graphql(meta.schema, findQueryByTitle(title));
      expect(result).toMatchSnapshot();
    });
  });

  {
    const title = 'Create user mutation';
    it(title, async () => {
      const result = await graphql(meta.schema, findQueryByTitle(title));
      expect(result.data.userCreate.record).toMatchSnapshot();
      expect(result.data.userCreate.clientMutationId).toMatchSnapshot();
    });
  }
});
