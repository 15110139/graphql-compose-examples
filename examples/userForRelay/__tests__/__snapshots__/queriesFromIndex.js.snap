exports[`userForRelay > queries Create user mutation 1`] = `
Object {
  "age": 24,
  "contacts": Object {
    "email": "mail@example.com",
    "phones": Array [
      "111-222-333-444",
      "444-555-666-777",
    ],
  },
  "gender": "ladyboy",
  "name": "My Name",
}
`;

exports[`userForRelay > queries Create user mutation 2`] = `"1"`;

exports[`userForRelay > queries Relay Connection 1`] = `
Object {
  "data": Object {
    "userConnection": Object {
      "edges": Array [
        Object {
          "cursor": "eyJfaWQiOiI1N2JiNDRkZDIxZDJiZWZiN2NhM2YwMDEifQ==",
          "node": Object {
            "_id": "57bb44dd21d2befb7ca3f001",
            "id": "VXNlcjo1N2JiNDRkZDIxZDJiZWZiN2NhM2YwMDE=",
            "name": "User 1",
          },
        },
        Object {
          "cursor": "eyJfaWQiOiI1N2JiNDRkZDIxZDJiZWZiN2NhM2YwMDIifQ==",
          "node": Object {
            "_id": "57bb44dd21d2befb7ca3f002",
            "id": "VXNlcjo1N2JiNDRkZDIxZDJiZWZiN2NhM2YwMDI=",
            "name": "User 2",
          },
        },
        Object {
          "cursor": "eyJfaWQiOiI1N2JiNDRkZDIxZDJiZWZiN2NhM2YwMDMifQ==",
          "node": Object {
            "_id": "57bb44dd21d2befb7ca3f003",
            "id": "VXNlcjo1N2JiNDRkZDIxZDJiZWZiN2NhM2YwMDM=",
            "name": "User 3",
          },
        },
      ],
    },
  },
}
`;

exports[`userForRelay > queries Relay node 1`] = `
Object {
  "data": Object {
    "node": Object {
      "_id": "57bb44dd21d2befb7ca3f010",
      "age": 21,
      "gender": "female",
      "id": "VXNlcjo1N2JiNDRkZDIxZDJiZWZiN2NhM2YwMTA=",
      "name": "User 10",
    },
  },
}
`;
